corehq/apps/case_importer/tracking/filestorage.py

```diff
@@ -0,0 +1,109 @@
+from collections import namedtuple
+import os
+from tempfile import mkstemp
+import uuid
+from django.core.cache import caches
+from corehq.apps.case_importer.tracking.models import CaseUploadFileMeta
+from corehq.blobs import get_blob_db
+from corehq.blobs.util import random_url_id
+from corehq.util.files import file_extention_from_filename
+from dimagi.utils.decorators.memoized import memoized
+
+BUCKET = 'case_importer'
+
+
+class PersistentFileStore(object):
+    """
+    helper for writing and reading back arbitrary files to HIPAA-compliant storage
+    (Riak or filesystem, via BlobDB)
+
+    metadata is also stored in a django model given by meta_model,
+    which is returned by write_file
+
+    """
+    def __init__(self, bucket, meta_model):
+        """
+        :meta_model is a django model used to store meta info
+            must contain columns identifier, filename, length
+        """
+
+        self._bucket = bucket
+        self._db = get_blob_db()
+        self._meta_model = meta_model
+
+    def write_file(self, f, filename):
+        identifier = random_url_id(16)
```

Comment:
I believe the reason is that this function packs the same number of bits into fewer characters, and riak cares somewhat about key length, since uuids are usually given in hex, and this uses base64. I'm mostly just using what blobdb itself uses.